src/software-delivery-machine/cloudFoundrySoftwareDeliveryMachine.ts

```diff
@@ -27,18 +41,30 @@ export function cloudFoundrySoftwareDeliveryMachine(opts: { useCheckstyle: boole
                 CloudFoundryProductionDeployOnSuccessStatus,
             ],
         },
-        new GuardedPhaseCreator(HttpServicePhases, PushesToDefaultBranch, HasCloudFoundryManifest, PushToPublicRepo, MaterialChangeToJavaRepo),
+        new GuardedPhaseCreator(HttpServicePhases, PushesToDefaultBranch, IsMaven, IsSpringBoot,
```

Comment:
I think it's confusing to use varargs here, because it is hard to distinguish the phase creator from the guards. an array would be better.

(obviously not  blocking)